doc/modules/cassandra/pages/architecture/cql-on-accord.adoc

```diff
@@ -0,0 +1,612 @@
+or
+`https://github.com/apache/cassandra/blob/122f5300855d56131948575f80ce0594547c9040/src/java/org/apache/cassandra/service/accord/txn/TxnDataRangeValue.java[TxnDataRangeValue]`
+. `TxnDataKeyValue` doesn’t support merging because Accord only reads
+from a single replica, but `TxnDataRangeValue` does because the integer
+key for `TxnData` identifies the logical read in the transaction, but
+the actual execution of the range read could touch an arbitrary number
+of command stores covered by the range and each will produce their own
+`TxnDataRangeValue` for their portion of the read.
+
+=== Result
+
+`https://github.com/apache/cassandra-accord/blob/134df57677bbd5092994923a4dc2f15cd1d033d1/accord-core/src/main/java/accord/api/Result.java[Result]`
+is the interface for what is returned by `Query` and ends up being
+returned as the non-error result by Accord to the coordinator of a
+transaction. This is also implemented by `TxnData` for key read results
+and by `TxnRangeReadResult` for range reads.
+
+There is also `RetryNewProtocolResult` which can be returned by
+Cassandra’s integration with Accord during live migration. This retry
+error indicates that Accord determined the transaction’s execute time is
+in an epoch where Accord does not manage some or all of that data for
+read or write so the transaction should be retried on whatever system
+currently manages that data.
+
+=== Read
+
+`https://github.com/apache/cassandra-accord/blob/trunk/accord-core/src/main/java/accord/api/Read.java#L32[Read]`
+is where a transaction defines how data should be read during execution
+in order to return a result, and it will have its `read` method invoked
+along with specific keys to be read at command stores.
+
+A `Read` has to define all the keys it will access up front and needs to
+support `slice/intersecting/merge` so Accord can send only the relevant
+parts of a transactions reads to the command stores that are responsible
+for persisting metadata about the transaction and executing the read.
+
+`https://github.com/apache/cassandra/blob/122f5300855d56131948575f80ce0594547c9040/src/java/org/apache/cassandra/service/accord/txn/TxnRead.java[TxnRead]`
+implements `Read` and is a sorted collection of
+`https://github.com/apache/cassandra/blob/122f5300855d56131948575f80ce0594547c9040/src/java/org/apache/cassandra/service/accord/txn/TxnNamedRead.java#L77[TxnNamedRead]`.
+The name in `TxnNamedRead` refers to what is now the integer identifier
+for each logical read in the transaction. `TxnNamedRead` supports both
+key and range reads although not both in the same transaction.
+
+The name for a read is an incrementing integer encoded at planning time
+and includes:
+
+* USER - let statements
+* RETURNING - Returning select in `TransactionStatement`
+* AUTO++_++READ - Automatically generated reads like list index set
+* CAS++_++READ - Read for CAS statements
```

Comment:
I'm not sure what this means.

Would examples of names be `123-USER`, `124-RETURNING`, `125-AUTO_READ`, `126-CAS_READ`, or are the types of things that it includes additive like `123-USER-RETURNING? `